tracer/src/Datadog.Trace/Debugger/Expressions/ProbeExpressionEvaluator.cs

```diff
@@ -24,16 +24,20 @@ internal class ProbeExpressionEvaluator
 
     private Lazy<CompiledExpression<double>?> _compiledMetric;
 
+    private Lazy<KeyValuePair<CompiledExpression<bool>, CompiledExpression<string>[]>[]> _compiledDecorations;
+
     private int _expressionsCompiled;
 
     internal ProbeExpressionEvaluator(
         DebuggerExpression[] templates,
         DebuggerExpression? condition,
-        DebuggerExpression? metric)
+        DebuggerExpression? metric,
+        KeyValuePair<DebuggerExpression, DebuggerExpression[]>[] spanDecorations)
```

Comment:
Why is it a KeyValuePair and not `DebuggerExpression[]`? 🤔 
Is the Key represents the expression of TagName and value represents the expression of TagValue? If that's the case, why isn't it simply `KeyValuePair<DebuggerExpression, DebuggerExpression>[]`?

Reply:
We discussed about it offline